survival_lib.py

```
@@ -80,6 +91,158 @@ def _sort_key(v: Any) -> Tuple[int, Union[float, str]]:
     return sorted(groups, key=_sort_key)
 
 
+def _extract_scalar(val: Any) -> float:
```

Ruff (0.14.10)

Dynamically typed expressions (typing.Any) are disallowed in `val`

(ANN401)